modules/dkimPolicy.jsm

```diff
@@ -409,6 +409,162 @@ var Policy = {
 		}
 	},
 
+	/**
+	 * Check that the list of signed headers satisfy our policies.
+	 * - Warn if recommended headers are not signed.
+	 * - Try detecting maliciously added unsigned headers.
+	 *
+	 * The detection for maliciously added unsigned headers 
+	 * only considers all the recommended signed headers.
+	 * It simply ensures that a message does not contain both 
+	 * signed and unsigned values of a header.
+	 * Using the same mechanism for all signed headers 
+	 * will cause problems there added headers is normal
+	 * e.g. the Received header.
+	 *
+ 	 * @param {MAP<String,String>} msgHeaders
+	 *			All headers of the message to check as headername-value-pair
+	 * @param {Object} DKIMSignature
+	 *			DKIMSignature of the message
+	 * @returns {void}
+	 */	
+	checkHeadersSigned: function Policy_checkHeadersSigned(msgHeaders, DKIMSignature) {
+		
+		const POLICY_DKIM_UNSIGNED_HEADERS_WARNING_MODE = {
+			RELAXED : 10,
+			RECOMMENDED : 20,
+			STRICT : 30
+		}
+		
+		// The list of recommended headers to sign is mostly based on
+		// https://www.rfc-editor.org/rfc/rfc6376.html#section-5.4.
+		const SIGNEDHEADERS = {
+			REQUIRED : ["From", "Subject"],
+			RECOMMENDED : ["Date", "To", "Cc", "Resent-Date", "Resent-From", "Resent-To", "Resent-Cc", "In-Reply-To", "References", "List-Id", "List-Help", "List-Unsubscribe", "List-Subscribe", "List-Post", "List-Owner", "List-Archive"],
+			DESIRED : ["Message-ID", "Sender", "MIME-Version", "Content-Transfer-Encoding", "Content-Disposition", "Content-ID", "Content-Description", "Reply-To"]
+		};
+	
+		// We would like Reply-To to be in the recommended list.
+		// As some bigger domains violate this, we will only enforce it if the Reply-To is not in the signing domain, but a valid email address
+		const replyTo = msgHeaders.get("reply-to");
+		let replyToAddress = null;
+		if (replyTo && replyTo[0]) {
+						
+			// Copied over from ARHParser.jsm
```

Comment:
It's ok for me if you want to start backporting also the parsing code for email addresses.
But note that in this older TB version it could be easier to reuse the TB internals like it is done for the from address.
https://github.com/lieser/dkim_verifier/blob/df20ce9beaca8bda79772b8cfad29d3765c27b44/chrome/content/dkim.js#L658-L660
Note that I did not test this for the reply to header.

Reply:
Thank you for pointing me at the easier and better way :)